### ISSUE TITLE
Fields in oneof not using TypeMapper

### ISSUE THREAD

**Issue by andrew-teirney** [scalapb/scalapb-validate#250]
I have noticed while adding `scalapb-validate` to my project that the code emitted by the code generator does not appear to use the `TypeMapper` for fields that are within a oneof.

A very simple protobuf that exhibits the problem is below

```syntax = "proto3";

package test;

import "google/protobuf/wrappers.proto";
import "scalapb/scalapb.proto";
import "validate/validate.proto";

message TestMessage {
  TestBigDecimal foo = 1 [(scalapb.field).type = "_root_.scala.math.BigDecimal"];
  oneof value {
    TestBigDecimal bar = 2 [(scalapb.field).type = "_root_.scala.math.BigDecimal"];
  }
}

message TestBigDecimal {}
```

The code emitted code is as follows.

```package test.test

object TestMessageValidator extends scalapb.validate.Validator[test.test.TestMessage] {
  def validate(input: test.test.TestMessage): scalapb.validate.Result =
    scalapb.validate.Result.optional(input.foo.map(test.test.TestMessage._typemapper_foo.toBase)) { _value =>
      test.test.TestBigDecimalValidator.validate(_value)
    } &&
    scalapb.validate.Result.optional(input.value.bar) { _value =>
      test.test.TestBigDecimalValidator.validate(_value)
    }
  
}
```
One can see from the above code that for the bar field, the type mapper has not been used to convert to the base type.

I believe this might be a simple bug in the creation of the accessor in the code referenced below. I believe the bug is that in the final `else` case (line 117) it does not invoke `toBase`, but I am not sure.

https://github.com/scalapb/scalapb-validate/blob/16f75aadae6a71abb79c4aeb24f5531a575287a4/code-gen/src/main/scala/scalapb/validate/compiler/CodeGenerator.scala#L110-L120 

It could be that I am just doing something wrong as well.

**Comment by xuwei-k** [comment]
Could you release new version with this fix? 🙏

**Comment by thesamet** [comment]
Released v0.3.4

**Comment by xuwei-k** [comment]
Thanks!
